backend/ee/danswer/server/query_and_chat/chat_backend.py

```diff
@@ -60,6 +63,27 @@ def translate_doc_response_to_simple_doc(
     ]
 
 
+def _get_final_context_doc_indices(
+    final_context_docs: list[LlmDoc] | None,
+    simple_search_docs: list[SimpleDoc] | None,
+) -> list[int] | None:
+    """
+    this function returns a list of indices of the simple search docs
+    that were actually fed to the LLM.
+    """
+    if final_context_docs is None or simple_search_docs is None:
+        return None
+
+    final_context_indices: list[int] = []
```

Comment:
Another super nit: set + list comprehension would be slightly faster (doesn't matter really since we'll only ever have a small amount of docs to check)
Something like 
```
    final_context_doc_ids = {all docs ids}
    return [i for i, doc in enumerate(simple_search_docs) if doc.id in final_context_doc_ids]
```